tests/test_flags.py

```diff
@@ -0,0 +1,294 @@
+        ins = test_flags(one=True, two=False)
+        other = test_flags(one=False, two=True)
+
+        third = ins
+        ins ^= other
+        assert ins is third
+        assert ins.value == 0b011
+
+        ins.two = False
+        other.one = True
+        ins ^= other
+        assert ins.value == 0b010
+
+    def test__le__(self, test_flags: Type[TestFlags]) -> None:
+        ins = test_flags(one=True, two=False)
+        other = test_flags(one=False, two=True)
+
+        assert not ins <= other
+        other.one = True
+        assert ins <= other
+
+        with pytest.raises(TypeError, match="cannot compare TestFlags with int"):
+            _ = ins <= 4  # type: ignore
+
+    def test__ge__(self, test_flags: Type[TestFlags]) -> None:
+        ins = test_flags(one=True, two=False)
+        other = test_flags(one=False, two=True)
+
+        assert not ins >= other
+        ins.two = True
+        assert ins >= other
+
+        with pytest.raises(TypeError, match="cannot compare TestFlags with int"):
+            _ = ins >= 4  # type: ignore
+
+    def test__lt__(self, test_flags: Type[TestFlags]) -> None:
+        ins = test_flags(one=True, two=False)
+        other = test_flags(one=False, two=True)
+
+        assert not ins < other
+        other.one = True
+        assert ins < other
+
+        with pytest.raises(TypeError, match="cannot compare TestFlags with int"):
+            _ = ins < 4  # type: ignore
+
+    def test__gt__(self, test_flags: Type[TestFlags]) -> None:
+        ins = test_flags(one=True, two=False)
+        other = test_flags(one=False, two=True)
+
+        assert not ins > other
+        ins.two = True
+        assert ins > other
+
+        with pytest.raises(TypeError, match="cannot compare TestFlags with int"):
+            _ = ins > 4  # type: ignore
+
+    def test__invert__(self, test_flags: Type[TestFlags]) -> None:
+        ins = test_flags(one=True)
+        assert ins.value == 0b0001
+        other = ~ins
+        assert ins.value == 0b0001
+        # the other `0` here is because invert does not invert values that are not defined
+        assert other.value == 0b10110
+
+    def test__hash__(self, test_flags: Type[TestFlags]) -> None:
+        ins = test_flags(one=True)
+        assert hash(ins) == hash(ins.value)
+
+    def test_iter(self, test_flags: Type[TestFlags]) -> None:
+        ins = test_flags(one=True, two=False)
+        ran_at_least_once = False
+        for flag, value in iter(ins):
+            ran_at_least_once = True
+            assert flag in ins.VALID_FLAGS
+            assert getattr(ins, flag) == value
+
+        assert ran_at_least_once
+
+    def test_from_value(self, test_flags: Type[TestFlags]) -> None:
+        ins = test_flags._from_value(0b101)
+        assert ins.value == 0b101
+
+    def test_has_flag(self, test_flags: Type[TestFlags]) -> None:
+        ins = test_flags()
+
+        ins.two = True
+
+        assert ins.two is True
```

Comment:
This is also covered by `test_set_and_get_flag` below

Reply:
I think I updated this

Reply:
Doesn't look like it:
https://github.com/DisnakeDev/disnake/blob/f39e229d7b12c00227f1f2e6084bf0381d68ef1c/tests/test_flags.py#L321-L322
I believe `test_has_flag` can just be removed

Reply:
https://github.com/DisnakeDev/disnake/pull/616/commits/312d762e43508ef8dd002bc6cb0c1d79196083aa